.github/workflows/_deploy_production.yml

```diff
@@ -0,0 +1,100 @@
+name: Deploy Production
+run-name: Triggered by ${{ github.actor }}
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: "Image tag to deploy. Defaults to the last commit SHA in the branch."
+        type: string
+        default: ${{ github.sha }}
+        required: false
+
+env:
+  # mark:automatic-version
+  VERSION: "1.0.0"
+
+jobs:
+  push:
+    permissions:
+      packages: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Login to staging registry
+        uses: ./.github/actions/gcp-docker-login
+        id: login-staging
+        with:
+          project: firezone-staging
+      - name: Login to production registry
+        uses: ./.github/actions/gcp-docker-login
+        id: login-production
+        with:
+          project: firezone-prod
+      - name: Pull and push images
+        run: |
+          set -xe
+          IMAGES=(relay api gateway web)
+          MAJOR_VERSION="${VERSION%%.*}"
+          MAJOR_MINOR_VERSION="${VERSION%.*}"
+
+          for image in "${IMAGES[@]}"; do
+            SOURCE_TAG=${{ steps.login-staging.outputs.registry }}/firezone/${image}:${{ inputs.tag }}
+            docker pull --platform linux/amd64 ${SOURCE_TAG}
+
+            echo "Retagging ${image} from ${SOURCE_TAG}"
+
+            docker tag ${SOURCE_TAG} ${{ steps.login-production.outputs.registry }}/firezone/${image}:${{ inputs.tag }}
+            docker tag ${SOURCE_TAG} ${{ steps.login-production.outputs.registry }}/firezone/${image}:${{ env.VERSION }}-${{ inputs.tag }}
+            docker tag ${SOURCE_TAG} ${{ steps.login-production.outputs.registry }}/firezone/${image}:${{ env.VERSION }}
+            docker tag ${SOURCE_TAG} ${{ steps.login-production.outputs.registry }}/firezone/${image}:${MAJOR_VERSION}
+            docker tag ${SOURCE_TAG} ${{ steps.login-production.outputs.registry }}/firezone/${image}:${MAJOR_MINOR_VERSION}
+
+            docker push --all-tags ${{ steps.login-production.outputs.registry }}/firezone/${image}
+          done
+
+  deploy-production:
+    needs: push
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    env:
+      TF_CLOUD_ORGANIZATION: "firezone"
+      TF_API_TOKEN: "${{ secrets.TF_API_TOKEN }}"
+      TF_WORKSPACE: "production"
+    steps:
+      - name: Get Terraform Version
+        run: |
+          TERRAFORM_VERSION=$(cat .tool-versions | grep terraform | awk '{ print $NF; }')
```

Comment:
We should use:
```
- name: Tool Versions
        id: versions
        uses: marocchino/tool-versions-action@v1.2.0
```

Reply:
Makes sense, I'll open that as a separate PR ontop of this one to keep this one from getting larger